File: baekjoon 문제집/ZOAC 4/Main.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int h = Integer.parseInt(st.nextToken());
        int w = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        
        int rowCnt = (h % (n + 1) != 0) ? (h / (n + 1) + 1) : (h / (n + 1));
        int colCnt = (w % (m + 1) != 0) ? (w / (m + 1) + 1) : (w / (m + 1));
        
        bw.write(rowCnt * colCnt + "\n");
        bw.flush();
        bw.close();
    }
}

File: baekjoon 문제집/ZOAC 4/Main.kt
import java.io.BufferedWriter
import java.io.OutputStreamWriter


fun main() = with(System.`in`.bufferedReader()) {
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val (h, w, n, m) = readln().split(' ').map { it.toInt() }

    val col = if (h % (1 + n) != 0) (h / (1 + n)) + 1 else h / (1 + n)
    val row = if (w % (1 + m) != 0) (w / (1 + m)) + 1 else w / (1 + m)

    bw.write("${col * row}")
    bw.flush()
    bw.close()
}
